src/tests/ftest/util/soak_test_base.py

```diff
@@ -294,6 +299,79 @@ def harasser_job_done(self, args):
+        jobid_list = []
+        node_list = self.hostlist_clients
+        lib_path = os.getenv("LD_LIBRARY_PATH")
+        path = os.getenv("PATH")
+        v_env = os.getenv("VIRTUAL_ENV")
+        env = ";".join([f"export LD_LIBRARY_PATH={lib_path}",
+                        f"export PATH={path}",
+                        f"export VIRTUAL_ENV={v_env}"])
+        for job_dict in self.joblist:
+            jobid_list.append(job_dict["jobid"])
+        self.log.info(f"Submitting {len(jobid_list)} jobs at {time.ctime()}")
+        jobs_not_done = jobid_list
+        while True:
+            if time.time() > self.end_time or len(jobid_list) == 0:
+                break
+            jobs = []
+            job_results = {}
+            for job_dict in self.joblist:
+                job_id = job_dict["jobid"]
+                if job_id in jobid_list:
+                    node_count = job_dict["nodesperjob"]
+                    if len(node_list) >= node_count:
+                        debug_logging(
+                            self.log, self.enable_debug_msg, f"DBG: node_count {node_count}")
+                        debug_logging(
+                            self.log,
+                            self.enable_debug_msg,
+                            f"DBG: node_list initial/queue {node_list}")
+                        job_node_list = node_list[:node_count]
+                        debug_logging(
+                            self.log,
+                            self.enable_debug_msg,
+                            f"DBG: node_list before launch_job {node_list}")
+                        script = job_dict["jobscript"]
+                        timeout = job_dict["jobtimeout"]
+                        log = job_dict["joblog"]
+                        error_log = job_dict["joberrlog"]
+                        method = launch_jobscript
+                        params = (self.log, job_queue, job_id, job_node_list,
+                                  env, script, log, error_log, timeout, self)
+                        name = f"SOAK JOB {job_id}"
+
+                        jobs.append(threading.Thread(
+                            target=method, args=params, name=name, daemon=True))
+                        jobid_list.remove(job_id)
+                        node_list = node_list[node_count:]
+                        debug_logging(
+                            self.log,
+                            self.enable_debug_msg,
+                            f"DBG: node_list after launch_job {node_list}")
+            # run job scripts on all available nodes
+            for job in jobs:
+                job.start()
+            # debug_logging(self.log, self.enable_debug_msg, "DBG: all jobs started")
+            # for job in jobs:
+            #     job.join()
+            # debug_logging(self.log, self.enable_debug_msg, "DBG: all jobs joined")
+            while not job_queue.empty():
+                job_results = job_queue.get()
```

Comment:
So this submits potentially multiple jobs and doesn't schedule anymore jobs until all of those are complete. I think you'd rather submit and complete asynchronously? I could help with that improvement if needed. There are a few approaches we could take